src/commands/force/source/beta/status.ts

```diff
@@ -0,0 +1,181 @@
+    const wantsRemote =
+      (this.flags.remote as boolean) || (this.flags.all as boolean) || (!this.flags.local && !this.flags.all);
+
+    this.logger.debug(
+      `project is ${this.project.getPath()} and pkgDirs are ${this.project
+        .getPackageDirectories()
+        .map((dir) => dir.path)
+        .join(',')}`
+    );
+    const tracking = await SourceTracking.create({
+      org: this.org,
+      project: this.project,
+      apiVersion: this.flags.apiversion as string,
+    });
+    let outputRows: StatusResult[] = [];
+
+    if (wantsLocal) {
+      await tracking.ensureLocalTracking();
+      const localDeletes = tracking.populateTypesAndNames({
+        elements: await tracking.getChanges<ChangeResult>({ origin: 'local', state: 'delete', format: 'ChangeResult' }),
+        excludeUnresolvable: true,
+        resolveDeleted: true,
+      });
+
+      const localAdds = tracking.populateTypesAndNames({
+        elements: await tracking.getChanges<ChangeResult>({ origin: 'local', state: 'add', format: 'ChangeResult' }),
+        excludeUnresolvable: true,
+      });
+
+      const localModifies = tracking.populateTypesAndNames({
+        elements: await tracking.getChanges<ChangeResult>({ origin: 'local', state: 'modify', format: 'ChangeResult' }),
+        excludeUnresolvable: true,
+      });
+
+      outputRows = outputRows.concat(localAdds.flatMap((item) => this.statusResultToOutputRows(item, 'add')));
+      outputRows = outputRows.concat(localModifies.flatMap((item) => this.statusResultToOutputRows(item, 'changed')));
+      outputRows = outputRows.concat(localDeletes.flatMap((item) => this.statusResultToOutputRows(item, 'delete')));
+    }
+
+    if (wantsRemote) {
+      // by initializeWithQuery true, one query runs so that parallel getChanges aren't doing parallel queries
+      await tracking.ensureRemoteTracking(true);
+      const [remoteDeletes, remoteModifies] = await Promise.all([
+        tracking.getChanges<ChangeResult>({ origin: 'remote', state: 'delete', format: 'ChangeResult' }),
+        tracking.getChanges<ChangeResult>({ origin: 'remote', state: 'nondelete', format: 'ChangeResult' }),
+      ]);
+      outputRows = outputRows.concat(remoteDeletes.flatMap((item) => this.statusResultToOutputRows(item)));
+      outputRows = outputRows.concat(remoteModifies.flatMap((item) => this.statusResultToOutputRows(item)));
+    }
+
+    if (wantsLocal && wantsRemote) {
+      // keys like ApexClass__MyClass.cls
+      const conflictKeys = (await tracking.getConflicts()).map((conflict) => getKeyFromObject(conflict));
+      if (conflictKeys.length > 0) {
+        outputRows = outputRows.map((row) =>
+          conflictKeys.includes(getKeyFromStrings(row.type, row.fullName))
+            ? { ...row, state: `${row.state} (Conflict)` }
+            : row
+        );
+      }
+    }
+
+    // sort order is state, type, fullname
+    outputRows.sort((a, b) => {
```

Comment:
should this sort of output stuff be moved into a formatter?

Reply:
that would be more consistent with the rest of plugin-source.